src/subnet_rental_canister/src/canister.rs

```diff
@@ -500,26 +521,27 @@ pub async fn refund() -> Result<u64, String> {
     if guard_res.is_err() {
         return Err("Failed to acquire lock. Try again.".to_string());
     }
-    // does the caller have an active rental request?
+    // Does the caller have an active rental request?
     match get_rental_request(&caller) {
         Some(
             rental_request @ RentalRequest {
                 user,
+                initial_cost_icp: _,
                 refundable_icp,
+                locked_amount_icp: _,
                 locked_amount_cycles,
                 initial_proposal_id,
                 creation_date: _,
                 rental_condition_id: _,
                 last_locking_time: _,
-                lock_amount_icp: _,
             },
         ) => {
-            println!("Refund requested for user principal {:?}", &caller);
-            // Refund the remaining ICP on the SRC main subaccount to the user.
+            println!("Refund requested for user principal {}", &caller);
+            // Refund the remaining ICP on the SRC/user subaccount to the user.
             let res = refund_user(user, refundable_icp - DEFAULT_FEE, initial_proposal_id).await;
```

Comment:
Why can't be refund all remaining ICP?

There could be a gap between what is in the account and the refundable_icp.

Reply:
If there is a gap, we have a bug. All code that changes that amount must also change refundable_icp. 

I am not checking the balance here because it is an additional inter-canister call, and I have all necessary information right here in memory.

Reply:
Ok, fair enough.